src/error.ts

```diff
@@ -0,0 +1,53 @@
+import { STATUS_CODES } from 'node:http';
+import { format } from 'node:util';
+
+/**
+ * Base HTTP error class built with a message and a status code
+ * @extends Error
+ */
+export class ErrorHTTP extends Error {
+  status: number;
+
+  constructor(message?: string | number, status: number = 500) {
+    super();
+
+    if (!message) {
+      this.status = status;
+      this.message = STATUS_CODES[status] as string;
+    } else if (typeof message === 'number') {
+      this.status = message;
+      this.message = STATUS_CODES[this.status] as string;
+    } else {
+      this.message = message;
+      this.status = status;
+    }
+
+    Error.call(this, this.message);
+    Error.captureStackTrace(this, ErrorHTTP);
+  }
+}
+
+/**
+ * Generates a custom error object used for specific error handling.
+ * Extends the ErrorHTTP class.
+ *
+ * @param {String} code - error code you want to use, ex: RateLimitError
+ * @param {Number} status - the HTTP status code you want this error to throw
+ * @extends ErrorHTTP
+ * @returns {Class} - an error handling class
+ */
+export function fastErrorHTTP(code: string, status?: number) {
+  if (typeof code !== 'string' && typeof code !== 'number') {
+    throw new Error('code is required to be a string or number');
+  }
```

Comment:
the type of `code` for the function doesn't match the types that are checked against here

Reply:
Good callout - updated the type to allow `number` and a test in 846cd08 and 3fbee63